best-practices.md

```diff
@@ -0,0 +1,56 @@
+# Best practices when creating a workflow
+A workflow should be developed in accordance with the guidelines outlined in the [Serverless Workflow definitions][1] documentation.
+
+This document provides a summary of several additional rules and recommendations to ensure smooth integration with other applications, such as the [Backstage Orchestrator][2] UI.
+
+## Workflow output schema
+To effectively display the results of the workflow and any optional outputs generated by the user interface, or to facilitate the chaining of workflow executions, it is important for a workflow to deliver its output data in a recognized structured format as defined by the [WorkflowResult schema][3].
+
+The output meant for next processing should be placed under `data.result` property.
+
+```
+id: my-workflow
+version: "1.0"
+specVersion: "0.8"
+name: My Workflow
+start: ImmediatelyEnd
+extensions:
+  - extensionid: workflow-output-schema
+    outputSchema: schemas/workflow-output-schema.json
+states:
+  - name: ImmediatelyEnd
+     type: inject
+     data:
+       result:
+          completedWith: success
+          message: A human-readable description of the successful status. Or an error.
+          output:
+            - key: foo
+              value: Example string value produced on the output. This might be an input for a next workflow.
+          nextWorkflows:
+            - id: my-next-workflow-id
+              name: Next workflow name suggested if this is an assessment workflow
+    end: true
+```
+
+Then the `schemas/workflow-output-schema.json` can look like (referencing the [WorkflowResult schema][3]):
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "WorkflowResult",
+    "description": "Schema of workflow output",
+    "type": "object",
+    "properties": {
+        "result": {
+            "$ref": "workflow-result-schema.json",
```

Comment:
Should not it be `shared/schemas/workflow-result-schema.json`?

Reply:
IMO, it doesn't matter, it's an example only. The locations may vary.

Reply:
Done